### PR TITLE
Extra supply

### PR DESCRIPTION
**Pull request by FlanFlanagan**
This is the tackle issue #224 by artificially increasing demand invisibly to users. This ensures a "supply buffer".

### CODE REVIEW

**Review by robfairh — COMMENTED** [review]
Shouldn't it be 1+self.demand_std_dev/100 ?

File: d3ploy/solver.py [hunk 1]
```diff
@@ -172,3 +172,7 @@ def minimize_number_of_deployment(proto_commod, remainder):
         break
 
     return deploy_dict
+
+def decommission(agent, diff):
```
Comment (gwenchee) [inline d3ploy/solver.py:176]: whats this?

File: d3ploy/demand_driven_deployment_inst.py [hunk 2]
```diff
@@ -286,7 +279,7 @@ def calc_diff(self, commod, time):
         if time not in self.commodity_supply[commod]:
             self.commodity_supply[commod][time] = 0.0
         supply = self.predict_supply(commod)
-        demand = self.predict_demand(commod, time)
+        demand = self.predict_demand(commod, time) * self.demand_std_dev
```
Comment (robfairh) [inline d3ploy/demand_driven_deployment_inst.py]:
```suggestion
        demand = self.predict_demand(commod, time) *(1+self.demand_std_dev/100.0)
```

Comment (gwenchee) [inline d3ploy/demand_driven_deployment_inst.py]: I agree with roberto that it should be `1+self.demand_std_dev/100` or `1+self.demand_std_dev` depending on whether or not we tell the user to input a percentage

**Review by gwenchee — CHANGES_REQUESTED** [review]

File: d3ploy/demand_driven_deployment_inst.py (hunk 2 above)
Comment (katyhuff) [reply d3ploy/demand_driven_deployment_inst.py]: But -- be careful with division by integers. Python 2.7 can, for edge cases, behave badly with std_dev/100... divide by 100.0 for robustness.

File: d3ploy/demand_driven_deployment_inst.py [hunk 3]
```diff
@@ -135,13 +135,6 @@ class DemandDrivenDeploymentInst(Institution):
         default=0
     )
 
-    demand_std_dev = ts.Double(
```
Comment (katyhuff) [inline d3ploy/demand_driven_deployment_inst.py:138]: why don't we need this anymore?

Comment (FlanFlanagan) [reply d3ploy/demand_driven_deployment_inst.py:138]: It's a double variable. It was in the code twice, I think from an old PR.

Comment (robfairh) [reply d3ploy/demand_driven_deployment_inst.py:138]: It was defined twice

**Comment by FlanFlanagan** [comment]
Change to per commod with a decimal value

**Comment by gwenchee** [comment]
Looks good to me. Could you add the variable into the README before we merge? @FlanFlanagan

File: d3ploy/demand_driven_deployment_inst.py [hunk 4]
```diff
@@ -128,18 +128,12 @@ class DemandDrivenDeploymentInst(Institution):
...
-        doc="The standard deviation adjustment for the demand side.",
-        tooltip="The standard deviation adjustment for the demand side.",
-        uilabel="Demand Std Dev",
-        default=0
+    supply_buffer = ts.MapStringDouble(
+        doc="The percent above demand the supply should hit. In decimal" +
+            "form",
+        tooltip="Buffer Amount in percent decimal form.",
+        uilabel="Supply Buffer"
```
Comment (gwenchee) [inline d3ploy/demand_driven_deployment_inst.py:135]:
```suggestion
        uilabel="Supply Buffer",
```

File: d3ploy/solver.py (hunk 1 above)
Comment (gwenchee) [reply d3ploy/solver.py:176]: i dont think this should be included in this PR

File: tests/performance_tests/algorithm_performance_tests.py [hunk 5]
```diff
@@ -40,7 +40,7 @@
 
 ##### List of types of calc methods that are to be tested #####
 calc_methods = ["ma", "arma", "arch", "poly",
-                "exp_smoothing", "holt_winters", "fft"]
+                "exp_smoothing", "holt_winters", "fft", "sw_seasonal"]
```
Comment (gwenchee) [inline tests/performance_tests/algorithm_performance_tests.py:43]: i dont think this should be included in this PR

**Review by gwenchee — CHANGES_REQUESTED** [review]

File: d3ploy/demand_driven_deployment_inst.py (hunk 4 above)
Comment (gwenchee) [inline d3ploy/demand_driven_deployment_inst.py:135]:
```suggestion
        alias=['supply_buffer', 'commod', 'buffer'],
        uilabel="Supply Buffer"
```

**Comment by gwenchee** [comment]
I pulled this branch into my computer and fixed all the issues that I commented on. Would you prefer if I PR into your branch or make a new PR into arfc/d3ploy's master branch @FlanFlanagan

**Comment by katyhuff** [comment]
@FlanFlanagan it looks like these commits have been incorporated into  #233 and improvements are being made there. Closing this issue.

**Pull request closed without merging**